--- a/scripts/netifd-wireless.sh	2023-04-14 18:04:28.144283331 -0700
+++ b/scripts/netifd-wireless.sh	2023-04-14 18:09:16.597854113 -0700
@@ -327,7 +327,7 @@
 	json_get_vars isolate proxy_arp
 
 	[ ${isolate:-0} -gt 0 -o -z "$network_bridge" ] && return
-	[ ${multicast_to_unicast:-1} -gt 0 -o ${proxy_arp:-0} -gt 0 ] && json_add_boolean isolate 1
+	[ ${multicast_to_unicast:-0} -gt 0 -o ${proxy_arp:-0} -gt 0 ] && json_add_boolean isolate 1
 }
 
 for_each_interface() {
